### ISSUE TITLE
panic: Fail in goroutine after TestFanoutSyncPoolE2E has completed

### ISSUE THREAD

**Issue by ian-mi** [google/knative-gcp#1084]
**Describe the bug**
After upgrading the cloudevents SDK to v2.0.0-rc4, TestFanoutSyncPoolE2E fails with the following error:
```
panic: Fail in goroutine after TestFanoutSyncPoolE2E has completed

goroutine 6762 [running]:
testing.(*common).Fail(0xc0002a8c60)
	/usr/lib/go/src/testing/testing.go:613 +0x11a
testing.(*common).Fail(0xc000c81e60)
	/usr/lib/go/src/testing/testing.go:607 +0x92
testing.(*common).Errorf(0xc000c81e60, 0xfe48e7, 0x22, 0xc0083dfc90, 0x2, 0x2)
	/usr/lib/go/src/testing/testing.go:712 +0x90
github.com/google/knative-gcp/pkg/broker/handler/pool/testing.assertEvent(0xc000c81e60, 0xc0006d0d40, 0x0, 0xc000660000, 0x61)
	/home/ian/go/src/github.com/google/knative-gcp/pkg/broker/handler/pool/testing/helper.go:493 +0x184
github.com/google/knative-gcp/pkg/broker/handler/pool/testing.(*Helper).VerifyNextTargetRetryEvent.func1(0xc000c81e60, 0xc0006d0d40, 0xc003908ee0, 0xc003699080, 0x52)
	/home/ian/go/src/github.com/google/knative-gcp/pkg/broker/handler/pool/testing/helper.go:444 +0xc5
github.com/google/knative-gcp/pkg/broker/handler/pool/testing.(*Helper).VerifyNextTargetRetryEvent(0xc0001a47c0, 0x1169820, 0xc0001a43c0, 0xc000c81e60, 0xc003699080, 0x52, 0xc0006d0d40)
	/home/ian/go/src/github.com/google/knative-gcp/pkg/broker/handler/pool/testing/helper.go:462 +0x59c
created by github.com/google/knative-gcp/pkg/broker/handler/pool.TestFanoutSyncPoolE2E.func5
	/home/ian/go/src/github.com/google/knative-gcp/pkg/broker/handler/pool/fanout_test.go:212 +0x1c4
exit status 2
FAIL	github.com/google/knative-gcp/pkg/broker/handler/pool	17.042s
```

 **Expected behavior**
The test should wait for all helpers to complete before finishing to prevent this type of error.

 **To Reproduce**

1. Unpin the cloudevents SDK and upgrade to v2.0.0-rc4.
2. Run TestFanoutSyncPoolE2E.

**Comment by grantr** [comment]
@ian-mi are you working on understanding the sdk upgrade blockers?

**Comment by ian-mi** [comment]
Yes, these test related issues are simply race conditions in the test which were obscuring the underlying root cause. The first issue was #1086 which was fixed by PR #1087. The second issue was an upstream SDK issue: https://github.com/cloudevents/sdk-go/issues/509. I have fixed this upstream and included a workaround in the SDK upgrade PR #1080 which is now ready to review. Once that is merged I will create a PR to fix this and similar race conditions in the fanout tests.
